zkvote/service/manager/manager.go

```diff
@@ -556,10 +559,19 @@ func (m *Manager) newAVoter(sub *subject.Subject, idc string) (*voter.Voter, err
 	}
 
 	utils.LogInfof("Register, subject:%s, id:%v", sub.HashHex().String(), idc)
-	_, err = voter.Register(id.NewIdentity(idc))
+	identity := id.NewIdentity(idc)
+	// Insert idenitty to identity pool
+	_, err = voter.Insert(identity)
+	if nil != err {
+		return nil, err
+	}
+
+	// Publish identity to other pubsub peers
+	err = voter.Join(identity)
```

Comment:
should hide `Join` in voter. `Voter` is responsible for a pub/sub network, should let `Voter` control when to publish identities. could move in next phase.